.github/workflows/swift-toolchain.yml

```diff
@@ -245,11 +250,20 @@ jobs:
           show-progress: false
 
       - uses: compnerd/gha-setup-vsdevenv@main
+        if: inputs.builder_os == 'Windows'
         with:
-          host_arch: amd64
+          host_arch: ${{ inputs.builder_arch }}
           components: 'Microsoft.VisualStudio.Component.VC.Tools.x86.x64;Microsoft.VisualStudio.Component.VC.Tools.ARM64'
           arch: ${{ matrix.arch }}
 
+      - uses: maxim-lobanov/setup-xcode@v1
+        if: inputs.builder_os == 'Darwin'
+        with:
+          xcode-version: latest-stable
+
+      - uses: seanmiddleditch/gha-setup-ninja@master
```

Comment:
Likewise.

Reply:
Don't create a new action - this one is fine to use. Just that before we can merge this, we will need to configure the permissions.

Reply:
It worked when I tried it because the downstream repo is configured to accept all actions. We should change that. I don't have access to this repo configuration so I don't know if it works here.

Reply:
Oh, if downstream has that ... let it fly. I'm less concerned about that restriction on upstream and more on downstream.

Reply:
I restored the ninja action.